external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -3003,6 +3003,17 @@ fn exp(context: &mut Context, pe: Box<P::Exp>) -> Box<E::Exp> {
     }
     let e_ = match pe_ {
         PE::Unit => EE::Unit { trailing: false },
+        PE::Parens(pe) => {
+            if let PE::Cast(_, _) = &pe.value {
+                let sp!(pe_loc, PE::Cast(plhs, pty)) = *pe else {
+                    unreachable!()
+                };
+                let e_ = exp_cast(context, /* in_parens */ true, plhs, pty);
+                return Box::new(sp(pe_loc, e_));
+            } else {
+                return exp(context, pe);
+            }
+        }
```

Comment:
```suggestion
        PE::Parens(pe) => {
             match pe {
                  sp!(pe_loc, PE::Cast(plhs, pty)) => {
                      let e_ = exp_cast(context, /* in_parens */ true, plhs, pty);
                      return Box::new(sp(pe_loc, e_));
                  }
                  pe => return exp(context, pe);
             }
        }
```

Reply:
Doesn't quite help because of the box, but I'll switch it anyway and hope Rust compiles away the unbox/rebox